test-data/unit/check-overloading.test

```diff
@@ -5001,3 +5001,18 @@ def f(x):
 
 reveal_type(f(g([])))  # E: Revealed type is 'builtins.list[builtins.int]'
 [builtins fixtures/list.pyi]
+
+[case testTypeCheckOverloadCoroutine]
+from types import coroutine
+from typing import overload
+@overload
+@coroutine
+def f(x: int) -> None: ...
+@overload
+@coroutine
+def f(x: str) -> None: ...
+def f(x): pass
+
+reveal_type(f) # E: Revealed type is 'Overload(def (x: builtins.int) -> typing.AwaitableGenerator[Any, Any, Any, None], def (x: builtins.str) -> typing.AwaitableGenerator[Any, Any, Any, None])'
```

Comment:
Could you add test that show that the overload actually works as expected? For example, test that `await f(1)` works, but `await f([])` does not.

Reply:
One more request for more test cases: could you try checking to see what happens when you try using `@abstractmethod` with `@coroutine`? (E.g. make sure we still get the correct types, make sure the result makes sense no matter which decorator you apply first)